Dashboard/app/js/lib/controllers/survey-controllers.js

```diff
@@ -977,6 +960,69 @@ FLOW.questionControl = Ember.ArrayController.create({
     }
   }.observes('FLOW.selectedControl.selectedQuestion'),
 
+  reorderQuestions: function (qgId, reorderPoint, reorderDirection) {
+    var questionsInGroup = FLOW.store.filter(FLOW.Question, function (item) {
+      return item.get('questionGroupId') == qgId;
+    });
+
+    // move items up to make space
+    questionsInGroup.forEach(function (item) {
+      if (reorderDirection == "down") {
+        if (item.get('order') > reorderPoint) {
+          item.set('order', item.get('order') + 1);
+        }
+      } else if (reorderDirection == "up") {
+        if (item.get('order') > reorderPoint) {
+          item.set('order', item.get('order') - 1);
+        }
+      }
+    });
+  },
+
+  submitBulkQuestionsReorder: function (qgIds) {
+    this.set('bulkCommit', true);
+    for (var i=0; i>qgIds.length; i++) {
+      var questionsInGroup = FLOW.store.filter(FLOW.Question, function (item) {
+        return item.get('questionGroupId') == qgIds[i];
+      });
+      // restore order in case the order has gone haywire
+      FLOW.questionControl.restoreOrder(questionsInGroup);
```

Comment:
given that the scope here is the `FLOW.questionControl` I suggest shortening -> `this.restoreOrder()`